platforms/documentation/docs/src/docs/userguide/dep-man/basics/variant_aware_resolution.adoc

```diff
@@ -0,0 +1,142 @@
+// Copyright (C) 2023 Gradle, Inc.
+//
+// Licensed under the Creative Commons Attribution-Noncommercial-ShareAlike 4.0 International License.;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://creativecommons.org/licenses/by-nc-sa/4.0/
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+[[variant_aware_resolution]]
+= 7. Variant Aware Dependency Resolution
+
+In Gradle, the traditional method of sharing artifacts between projects involves the producer defining a configuration, and the consumer explicitly stating which configuration to depend on.
+However, this method isn't ideal when you want to automate the selection of *variants* based on the consumer's needs without specifying a particular configuration.
+
+Gradle allows for a more dynamic approach called **variant-aware resolution**, where both the producer and consumer define their requirements using **attributes**.
```

Comment:
```suggestion
In Gradle, dependency resolution is often thought of from the standpoint of a *consumer* and a *producer*.

The consumer declares dependencies and performs dependency resolution, while producers satisfy those dependencies by exposing variants.

Gradle's resolution engine follows a dynamic approach to dependency resolution called **variant-aware resolution**, where the consumer defines requirements using **attributes**, which are matched with the attributes declared by the producer.
```

I don't even think we should talk about this "traditional" method. If we want to mention it at the end we can but I would rather not lead users astray with this

Reply:
Done.